### PR TITLE
Fix #480: OIDC: Implement end-to-end tests

### LINKED ISSUE #480: OIDC: Implement end-to-end tests
Implement end-to-end tests for activation using OIDC.

### PR DESCRIPTION
**Pull request by banterCZ**
- [x] Depends on https://github.com/wultra/powerauth-restful-integration/pull/547

### CODE REVIEW

File: powerauth-backend-tests/src/test/resources/application.properties [hunk 1]
```diff
@@ -35,5 +35,14 @@ powerauth.test.identity.otp-verification.skip=false
 # which pass all checks, and there exists a workaround for the iProov verification process in tests.
 powerauth.test.identity.result-verification.skip=false
 
+# OIDC activation
+powerauth.test.activation.oidc.username=
+powerauth.test.activation.oidc.password=
+powerauth.test.activation.oidc.issuerUrl=
+powerauth.test.activation.oidc.clientId=
+powerauth.test.activation.oidc.redirectUri=
+powerauth.test.activation.oidc.providerId=
```
Comment (banterCZ) [inline powerauth-backend-tests/src/test/resources/application.properties:44]: - [ ] We should run the test only if then only if the property is filled.

**Review by romanstrobl — APPROVED** [review]
Good job.

I suggest two future improvements:
- test the application configuration endpoint, too, e.g. `scope` is fixed value in the test
- make sure the test runs out of the box without configuration
  - if we cannot use an existing OAuth2 authorization server, the tests can also run with mock server

File: .github/workflows/maven-integration-test.yml [hunk 2]
```diff
@@ -44,6 +44,13 @@ jobs:
           POWERAUTH_SERVICE_SECURITY_CLIENTTOKEN: ${{ secrets.POWERAUTH_SERVICE_SECURITY_CLIENTTOKEN }}
           POWERAUTH_SERVICE_SECURITY_CLIENTSECRET: ${{ secrets.POWERAUTH_SERVICE_SECURITY_CLIENTSECRET }}
           POWERAUTH_TEST_INCLUDECUSTOMTESTS: ${{ inputs.includeCustomTests == '' || inputs.includeCustomTests }} # default includeCustomTests=true even for 'schedule' event
+          POWERAUTH_TEST_ACTIVATION_OIDC_USERNAME: ${{ secrets.POWERAUTH_TEST_ACTIVATION_OIDC_USERNAME }}
```
Comment (banterCZ) [inline .github/workflows/maven-integration-test.yml:47]: - [ ] @zcgandcomp Please fill these secrets. I will pass them via 1password.

File: powerauth-backend-tests/src/main/java/com/wultra/security/powerauth/configuration/PowerAuthOidcActivationConfigurationProperties.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+ */
+@ConfigurationProperties(prefix = "powerauth.test.activation.oidc")
+@Component
+@Getter
+@Setter
+public class PowerAuthOidcActivationConfigurationProperties {
+
+    private String username;
+    private String password;
+    private String issuerUrl;
```
Comment (banterCZ) [inline powerauth-backend-tests/src/main/java/com/wultra/security/powerauth/configuration/PowerAuthOidcActivationConfigurationProperties.java:38]: 🤔 Let's limit the duplicity and take values from db (whatever is possible).

File: .github/workflows/maven-integration-test.yml (hunk 2 above)
Comment (zcgandcomp) [reply .github/workflows/maven-integration-test.yml:47]: DONE

**Pull request merged**